kanidm: Fix website and pam/nss in package

Flakebi opened the pull request:
Also fix some bugs in the module and improve the test

Flakebi committed:
kanidm: Fix website and pam/nss in package

Also fix some bugs in the module and improve the test

mweinelt commented on nixos/modules/services/security/kanidm.nix:
```diff
@@ -199,10 +200,32 @@ in
+              -n ${cfg.ensureDomainName}
+            ${pkgs.coreutils}/bin/rm "$config"
+          fi
         '';
```
Using `systemd.services.kanidm.path = with pkgs; [ coreutils  gnugrep gnused ];` would make this script look less mutilated! :laughing:

mweinelt commented on nixos/tests/kanidm.nix:
```diff
@@ -16,24 +12,41 @@ import ./make-test-python.nix ({ pkgs, ... }:
+              proxy_http_version  1.1;
+              proxy_cache_bypass  $http_upgrade;
+
+              proxy_set_header Upgrade           $http_upgrade;
+              proxy_set_header Connection        "upgrade";
+              proxy_set_header Host              $host;
+              proxy_set_header X-Real-IP         $remote_addr;
+              proxy_set_header X-Forwarded-For   $proxy_add_x_forwarded_for;
+              proxy_set_header X-Forwarded-Proto $scheme;
+              proxy_set_header X-Forwarded-Host  $host;
+              proxy_set_header X-Forwarded-Port  $server_port;
```
These lines boil down to `recommendedProxySettings` and `proxyWebsockets`.

mweinelt commented on nixos/tests/kanidm.nix:
```diff
@@ -42,17 +55,17 @@ import ./make-test-python.nix ({ pkgs, ... }:
         uri = "https://${serverDomain}"
-        verify_ca = true
-        verify_hostnames = true
       '';
```
What does this configure? Shouldn't this part be configured through the kanidm client module bits?

Flakebi replied:
There’s not really a client module at this moment, so the config file only gets created if a server or pam support is enabled :/

Oh, and these verify settings are `true` by default, so I removed them.

mweinelt replied:
Yeah, so I wonder why we need this in the test instead of letting the module create it.

Flakebi replied:
I split the options into `enableClient` and `enableServer`, so the test can use `clientSettings` now.

Flakebi committed:
kanidm: Simplify test and StartPre cmd

Flakebi committed:
kanidm: split settings for client and server

Have a separate enableClient and enableServer, so that clientSettings
can be used without running a server.

erictapen commented:
And again I didn't look properly before working on the branch ;)
I fixed the `ldapbindaddress` issue by filtering null values in 964ec52d954a7d3f666769ae8e32b29223dad550. This way we are able to document the option. Could you rebase ontop of that?

mweinelt commented on nixos/modules/services/security/kanidm.nix:
```diff
@@ -189,20 +198,43 @@ in
+              -c "$config" \
+              -n ${cfg.ensureDomainName}
+            rm "$config"
+          fi
```
Please present that hack upstream so it gets proper recognition. :facepalm: 

Anyway, it works!

erictapen replied:
I'm really hesitant about that amount of shell magic in init files. Would really prefer that logic to be handled by upstream. But as it is not user facing, I'd be okay with it if it works.

Flakebi replied:
I can see what you mean :D I mentioned it in https://github.com/kanidm/kanidm/issues/623

mweinelt commented:
![image](https://user-images.githubusercontent.com/131599/153720294-365ba4fb-4095-4f26-9fc3-ec1f4b336cf2.png)


:partying_face:
